headless-services/spring-boot-language-server/src/main/java/org/springframework/ide/vscode/boot/java/commands/JsonNodeHandler.java

```diff
@@ -97,17 +98,22 @@ public void handleStereotype(Stereotype stereotype, NodeContext context) {
 			for (Object source : sources) {
 				if (source instanceof URL) {
 					URL url = (URL) source;
-					if (url.getProtocol().equals("jar")) {
-						reference = convertUrlToJdtUri((URL) source, project.getElementName());
-					}
-					else if (url.getProtocol().equals("file")) {
-						try {
-							reference = url.toURI().toASCIIString();
-						}
-						catch (URISyntaxException e) {
-							// something went wrong
-						}
-					}
+					reference = ProjectBasedCatalogSource.getDefaultStereotypePath(url)
```

Comment:
I would vote for turning all the URLs into a `spring-boot-ls://resource` URL and then use that URL when reading the content of the resource (see below where we handle the command). I think we don't need to distinguish between URLs that are based on the default catalog files from inside of the language server and URLs that point to "random" JARs from the classpath - and therefore don't convert the JAR URLs into something JDT specific. We could just pass the URL to the client, then transport the URL back to the fetch content command, and read all the content from the URL - whether it is a file URL or a JAR URL.